runtime/doc/deprecated.txt

```diff
@@ -33,6 +33,8 @@ LSP
 • *vim.lsp.client_is_stopped()*		Use |vim.lsp.get_client_by_id()| instead.
 • *vim.lsp.util.stylize_markdown()*	Use |vim.treesitter.start()| with
 					`vim.wo.conceallevel = 2`.
+• *vim.lsp.log.should_log()*		Use |vim.lsp.log.set_format_func()| instead
+					and return `nil` to omit entries from the logfile.
```

Comment:
`should_log()` was more for plugins/user-code to check how a log-level compares to the current log-level. However, this definitely isn't lsp-specific so might as well deprecate this. If it's needed in the near-term, we may want to introduce `vim.log.cmp()` and/or `vim.log.has_level()`.